### PR TITLE
feat: [Week-2.6] Update, Translate, and Expand "HTTP/HTTPS and Web Security" Module

### PR DESCRIPTION
**Pull request by haslie22**
#### Title of Pull Request

Update, Translate, and Expand "HTTP/HTTPS and Web Security" Module

#### 🤔 This is a ...

- [ ] 🌟 New task
- [ ] 🌐 New module
- [ ] ⚙️ Update to an existing task
- [x] 🔧 Update to an existing module
- [x] 🔗 Update or addition of external resources or links
- [ ] 🐛 Fix in a task or related content
- [ ] 🛠 Fix in a module or related content
- [ ] ✏️ Fixed a typo or grammatical error
- [ ] 🔗 Fixed a broken link
- [ ] ❓ Other (specify: **\*\*\*\***\_\_\_\_**\*\*\*\***)

#### Description

- [x] Module description translated
- [x] Module topics and materials list expanded
- [x] Module materials revised and updated

#### Additional Information

Issue: [1490](https://github.com/orgs/rolling-scopes-school/projects/4/views/1?pane=issue&itemId=52830442)

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [ ] 📝 I have commented my code, particularly in hard-to-understand areas.
- [ ] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.

### CODE REVIEW

File: stage2/modules/http/README.md [hunk 1]
```diff
@@ -1,26 +1,83 @@
...
+- **CORS:**
+
+  - [Introducing CORS: Cross-Origin Resource Sharing](https://www.youtube.com/watch?v=JVZIhCVFJ9c)
+
+- **API Security Best Practices:**
+
+  - [OWASP Top 10 API Security Risks – 2023](https://owasp.org/API-Security/editions/2023/en/0x11-t10/)
+  - [The Fundamentals of Rate Limiting](https://medium.com/@patrikkaura/the-fundamentals-of-rate-limiting-how-it-works-and-why-you-need-it-fd86d39e358d)
+  - [Input Validation Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html)
+  - [Error Handling Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Error_Handling_Cheat_Sheet.html)
```
Comment (valerydluski) [inline stage2/modules/http/README.md:68]: We have a separate module on error handling

File: stage2/modules/http/README.md [hunk 2]
```diff
@@ -1,26 +1,83 @@
...
+  - [Authentication vs Authorization](https://www.freecodecamp.org/news/whats-the-difference-between-authentication-and-authorisation/)
+  - [Authorization Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Authorization_Cheat_Sheet.html)
+  - [Authentication Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html)
+  - [OAuth 2 Simplified](https://aaronparecki.com/oauth-2-simplified/)
+  - [Tutorial: How OAuth 2 works with real world example](https://www.youtube.com/watch?v=RROVLdfGmuE)
+  - [JSON Web Tokens (JWT) — the only explanation you will ever need](https://arielweinberger.medium.com/json-web-token-jwt-the-only-explanation-youll-ever-need-cf53f0822f50)
+
+- **Emerging Technologies and Future Trends:**
+
+  - [HTTP/3 From A To Z: Core Concepts](https://www.smashingmagazine.com/2021/08/http3-core-concepts-part1/)
```
Comment (valerydluski) [inline stage2/modules/http/README.md:82]: We already suggested studying this article as part of the module
https://github.com/rolling-scopes-school/tasks/tree/master/stage1/modules/client-server

File: stage2/modules/http/README.md [hunk 3]
```diff
@@ -1,26 +1,83 @@
...
+  - [The Road to QUIC](https://blog.cloudflare.com/the-road-to-quic)
```
Comment (valerydluski) [inline stage2/modules/http/README.md:83]: This is also more relevant for the client-server module

**Comment by valerydluski** [comment]
I think we can completely rename the module to Web Security, and focus on that. We recommended familiarizing with HTTP/HTTPS a bit earlier. If you have any questions, please ping me in private messages

**Review by KristiBo — APPROVED** [review]
Good job :thumbsup:

**Review by valerydluski — APPROVED** [review]
Excellent work on this PR! The changes you've made are fantastic, and it's clear you've put a lot of effort into ensuring everything is top-notch. Great job!

**Pull request merged**